lib/cn/cn_tree.c

```diff
@@ -1294,10 +1296,10 @@ cn_comp_update_kvcompact(struct cn_compaction_work *work, struct kvset *new_kvse
             le = tmp;
         }
 
-        if (new_kvset) {
+        if (new_kvset)
             kvset_list_add(new_kvset, &le->le_link);
-            work->cw_node->tn_cgen++;
-        }
+
+        work->cw_node->tn_cgen++;
```

Comment:
This bug kept us from noticing when a node went empty.  Well, before join it was the right call since there was nothing we could do with an empty node, but now we can remove it, huzzah!